Build failure with mingw

### Observed behavior

```console
x86_64-w64-mingw32-gcc: error: /W4: linker input file not found: No such file or directory
x86_64-w64-mingw32-gcc: warning: /wd4100: linker input file unused because linking not done
x86_64-w64-mingw32-gcc: error: /wd4100: linker input file not found: No such file or directory
x86_64-w64-mingw32-gcc: warning: /wd4200: linker input file unused because linking not done
x86_64-w64-mingw32-gcc: error: /wd4200: linker input file not found: No such file or directory
x86_64-w64-mingw32-gcc: warning: /wd4130: linker input file unused because linking not done
x86_64-w64-mingw32-gcc: error: /wd4130: linker input file not found: No such file or directory
x86_64-w64-mingw32-gcc: warning: /wd4127: linker input file unused because linking not done
x86_64-w64-mingw32-gcc: error: /wd4127: linker input file not found: No such file or directory
make[2]: *** [src/CMakeFiles/nats.dir/asynccb.c.obj] Error 1
make[2]: *** Deleting file `src/CMakeFiles/nats.dir/asynccb.c.obj'
make[2]: *** Waiting for unfinished jobs....
```

These compile flags can only be used with MSVC.
https://github.com/nats-io/nats.c/blob/7b5d3d735a62a8ff6e43f56b30efa4e51795cbce/CMakeLists.txt#L210-L215

### Expected behavior

Work.

### Server and client version

tag: v3.9.2

### Host environment

_No response_

### Steps to reproduce

_No response_

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -208,11 +208,16 @@ elseif(WIN32)
   set(NATS_PLATFORM_INCLUDE "win")
 
   # Warning control.
-  add_compile_options(/W4) # Set warning level to maximum, then disable:
-  add_compile_options(/wd4100) # unreferenced formal parameter
-  add_compile_options(/wd4200) # nonstandard extension used: zero-sized array in struct/union
-  add_compile_options(/wd4130) # logical operation on address of string constant
-  add_compile_options(/wd4127) # conditional expression is constant
+  if(MSVC)
+    # Set warning level to maximum, then disable:
+    # /wd4100 unreferenced formal parameter
+    # /wd4200 nonstandard extension used: zero-sized array in struct/union
+    # /wd4130 logical operation on address of string constant
+    # /wd4127 conditional expression is constant
+    add_compile_options(/W4 /wd4100 /wd4200 /wd4130 /wd4127)
+  else()
+    add_compile_options(-Wall -Wextra -Wno-unused-parameter -Wno-pedantic -Wno-address -Wno-constant-conditional)
+  endif()
 
   if(sodium_USE_STATIC_LIBS)
     add_definitions(

